server/apps/microservices/src/processors/machine-learning.processor.ts

```diff
@@ -45,24 +45,47 @@ export class MachineLearningProcessor {
     try {
       const { asset }: { asset: AssetEntity } = job.data;
 
-      const res = await axios.post(
-        immich_machine_learning_url + '/object-detection/detect-object',
-        {
-          thumbnailPath: asset.resizePath,
-        },
-      );
+      const res = await axios.post(immich_machine_learning_url + '/object-detection/detect-object', {
+        thumbnailPath: asset.resizePath,
+      });
 
       if (res.status == 201 && res.data.length > 0) {
-        const smartInfo = new SmartInfoEntity();
-        smartInfo.assetId = asset.id;
-        smartInfo.objects = [...res.data];
+        this.createAndSaveTagToAsset(asset, res.data);
+      }
+    } catch (e) {
+      this.logger.error(e);
+    }
+  }
+
+  private async createAndSaveTagToAsset(asset: AssetEntity, tagNames: string[]) {
+    try {
+      const immichTags: TagEntity[] = [];
+
+      for (const tag of tagNames) {
+        let immichTag: TagEntity | null;
+
+        immichTag = await this.tagRepository.findOne({ where: { name: tag, userId: asset.userId } });
+
+        if (!immichTag) {
+          immichTag = new TagEntity();
+          immichTag.name = tag;
+          immichTag.type = TagType.OBJECT;
+          immichTag.userId = asset.userId;
+
+          immichTag = await this.tagRepository.save(immichTag);
+        }
+
+        immichTags.push(immichTag);
+      }
+
+      const assetEntity = await this.assetRepository.findOne({ where: { id: asset.id } });
```

Comment:
Should we do this earlier, so we can skip potentially creating the tags if the asset doesn't exist?